code/game/machinery/doors/airlock.dm

```diff
@@ -197,6 +196,36 @@
 	if(id_tag)
 		id_tag = "[port.shuttle_id]_[id_tag]"
 
+/obj/machinery/door/airlock/proc/rebuild_parts()
+	if(part_overlays)
+		vis_contents -= part_overlays
+		QDEL_LIST(part_overlays)
```

Comment:
Why are these overlays in particular not built with the vis_overlays system? would need a unique flag but they seem like good candidates.

Reply:
vis_overlays overlay objects don't have the necessary vars